changelog/22583.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
```

Comment:
I think technically, we can changelog this as `bug`, since it's a performance regression from past versions that we're fixing.